addons/mail/static/src/widgets/form_renderer/form_renderer_tests.js

```diff
@@ -448,6 +457,204 @@ QUnit.test('chatter should become enabled when creation done', async function (a
     );
 });
 
+QUnit.test('Form view not scrolled when switching record', async function (assert) {
+    assert.expect(7);
+
+    this.data['res.partner'].records = [{
+        activity_ids: [],
+        id: 2,
+        display_name: "second partner",
+        description: [...Array(60).keys()].join('\n'),
+        message_ids: [],
+        message_follower_ids: [],
+    }, {
+        activity_ids: [],
+        id: 3,
+        display_name: "third partner",
+        message_ids: [],
+        message_follower_ids: [],
+    }];
+
+    const messages = [...Array(60).keys()].map(id => {
+        let partnerId;
+        if (id % 2 === 0) {
+            partnerId = 2;
+        } else {
+            partnerId = 3;
+        }
+        return {
+            author_id: [10, "Demo User"],
+            body: `<p>Message ${id + 1}</p>`,
+            date: "2019-04-20 10:00:00",
+            id: id + 1,
+            message_type: 'comment',
+            model: 'res.partner',
+            record_name: 'General',
+            res_id: partnerId,
+        };
+    });
+    this.data['mail.message'].records = messages;
+
+    await this.createView({
+        data: this.data,
+        hasView: true,
+        // View params
+        View: FormView,
+        model: 'res.partner',
+        arch: `
+            <form string="Partners">
+                <sheet>
+                    <field name="name"/>
+                    <field name="description"/>
+                </sheet>
+                <div class="oe_chatter">
+                    <field name="message_ids"/>
+                </div>
+            </form>
+        `,
+        viewOptions: {
+            currentId: 2,
+            ids: [2, 3]
+        },
+        config: {
+            device: {size_class: 4}
+        }
+    });
+
+    const controllerContentEl = document.querySelector('.o_content');
+    const formViewEl = document.querySelector('.o_form_view');
+    assert.doesNotHaveClass(formViewEl, 'o_xxl_form_view',
+        "Form view is not xxl"
+    );
+
+    assert.strictEqual(
+        document.querySelector('.breadcrumb-item.active').textContent,
+        'second partner',
+        "'second partner' form view is opened"
+    );
+    assert.strictEqual(controllerContentEl.scrollTop, 0,
+        "The initial scrollTop of the form view is 0"
+    );
+    // scroll message list to a big value to get to the bottom
+    let scrollRecord1;
+    await afterNextRender(() => {
+        controllerContentEl.scrollTop = 9000;
+        scrollRecord1 = controllerContentEl.scrollTop;
```

Comment:
yep, but aku asks for a real scroll to bottom:  `el.scrollTop = el.scrollHeight`
The spec of the scrollTop is that if the value is bigger than possible scroll, it takes the maximum

Reply:
My comment and @seb-odoo go in the same direction.
Hard-coded value is weak, because you cannot guarantee it really goes to bottom (what if bottom is at `10000`? This is style and browser dependent). Also exact move to bottom show confidence and exact expectation of what you want to test and how it should behave, resulting in a well-written and robust test.

(Note that `el.scrollTop = el.scrollHeight` is not correct, because `scrollHeight` corresponds to `scrollBottom` at the bottom of scrollable view)